### ISSUE TITLE
Consider making tree-sitter optional

### ISSUE THREAD

**Issue by tarsius** [erickgnavar/tree-sitter-ispell.el#3]
Since this package support built-in tree-sitter support, you should make the dependency on the `tree-sitter` package optional.

If you should declare all used functions and variables from that package, to make the compiler happy, even when the package is not available.

Alternatively you might want to remove support for `tree-sitter` altogether. This is what it's readme has to say:

> For Emacs 29+, please use [the built-in integration](https://lists.gnu.org/archive/html/emacs-devel/2022-11/msg01443.html) instead of this package.

**Comment by tarsius** [comment]
Note to self: Add to Emacsmirror again once there `tree-sitter` isn't a (hard) dependency anymore.

**Comment by erickgnavar** [comment]
Hi, I plan to support both versions(It partially does already) but I don't know how to manage that, on the context of being a MELPA installable package do you think it will be better to split this package in 2, one that support older versions of emacs and use `tree-sitter` and another that only support 29+ and use `treesit`?

**Comment by tarsius** [comment]
Yes, I think two packages (in two separate repositories) would be best. Benefits include:

- not having to declare *all* used functions and variables from *both* implementations, so that the compiler is happy when *the other* implementation is not available.
- easier distribution on Melpa and Emacsmirror
- users not going "it's named tree-sitter-*, so it is for the old `tree-sitter`, and since I am using the built-in implementation, there's nothing for me here"

**Comment by erickgnavar** [comment]
Thanks for your suggestions, I'm going to revert the new implementation from this repository and create a new one for `treesit`

**Comment by tarsius** [comment]
Thanks!

**Comment by erickgnavar** [comment]
Done, I made the changes to this repo, create a new one and make a pull request to MELPA to include the new one

https://github.com/melpa/melpa/pull/9060

**Comment by tarsius** [comment]
Thanks! 😉
